packages/rxjs-utils/src/lib/persistent-behavior-subject/persistent-behavior-subject.ts

```diff
@@ -0,0 +1,87 @@
+// persistent-behavior-subject
+
+import { BehaviorSubject } from 'rxjs';
+
+export type AcceptedStorage = { type: AcceptedStorageType, options: AcceptedStorageOption; };
+
+export type AcceptedStorageType = Storage | 'custom';
+export type AcceptedStorageOption = {
+  name?: string,
+  write?: (val: any) => void,
+  delete?: () => void;
+};
+
+/**
+ * A variant of BehaviorSubject that automatically stores the most recent value to storage.
+ * The default location is localStorage but can be specified to use anything.
+ *
+ * @class PersistentBehaviorSubject<T>
+ */
+export class PersistentBehaviorSubject<T> extends BehaviorSubject<T> {
+  public storage: AcceptedStorage;
+
+  constructor(value: T, storage?: AcceptedStorage | string) {
+    super(value);
+    if (typeof storage === 'string') {
+      this.storage = { type: localStorage, options: { name: storage } };
+    } else {
+      this.storage = storage ?? { type: localStorage, options: { name: 'PBS-' + Date.now() } };
+    }
+    this.write(value);
+  }
+
+  /**
+   * Save the data to the specified storage location.
+   */
+  private write(newValue: T): void {
+    switch (this.storage.type) {
```

Comment:
So...and, this doesn't need to be done in this PR, but more of a future enhancement. I was chatting with Stratton about this, and we were thinking this would be a good use case for the Strategy pattern. Instead of hard-coding options for storage, we could create an interface that can be implemented by the end developer, allowing them to plug in their own storage strategies. 

If you are not aware of the Strategy software design pattern, then enhancing this feature of our rxjs utils could be a great way to teach one of the more powerful design patterns available to us. Another good example of the strategy pattern is the Node.js auth library Passport, which uses strategies I think in a couple of ways.

Reply:
Our default strategy, bundled with the library, could be LocalStorage. We could maybe even include a handful of pre-made strategies, including Local Storage, Session Storage, and IndexedDB. A companion library might include an Ionic Storage strategy. The end developer could use one of our strategies, or implement their own.